applicationset/services/pull_request/bitbucket_server.go

```diff
@@ -51,9 +51,8 @@ func NewBitbucketServiceNoAuth(ctx context.Context, url, projectKey, repositoryS
 func newBitbucketService(ctx context.Context, bitbucketConfig *bitbucketv1.Configuration, projectKey, repositorySlug string, scmRootCAPath string, insecure bool, caCerts []byte) (PullRequestService, error) {
 	bitbucketConfig.BasePath = utils.NormalizeBitbucketBasePath(bitbucketConfig.BasePath)
 	tlsConfig := utils.GetTlsConfig(scmRootCAPath, insecure, caCerts)
-	bitbucketConfig.HTTPClient = &http.Client{Transport: &http.Transport{
-		TLSClientConfig: tlsConfig,
-	}}
+	bitbucketConfig.HTTPClient = &http.Client{Transport: http.DefaultTransport.(*http.Transport).Clone()}
+	bitbucketConfig.HTTPClient.Transport.(*http.Transport).TLSClientConfig = tlsConfig
```

Comment:
I think something like this reads more easily with only a single type assertion, what do you think? 
```suggestion
	transport := http.DefaultTransport.(*http.Transport).Clone()
	transport.TLSClientConfig = tlsConfig
	bitbucketConfig.HTTPClient = &http.Client{Transport: transport}
```

Reply:
agreed, implemented! and reordered the statements within this code as well.